### DISCUSSION TITLE
Type is required while creating new project

### DISCUSSION THREAD

**Discussion by andredev93** [mia-platform/community#270] [Q&A] [answered]
### Which section?

Console

### Which is the version?

10.8

### Description

Hi!

In an On-prem installation of the Console with Github provider we are having the error I copied below while creating a new project.
Can you please help me finding the issue?

Request:
```GET /api/backend/projects/draft?projectId=newprojectname&tenantId=tenant&providerId=github&projectName=newprojectname```
Status code: 500
Response body: 
```
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "\"type\" is required!"
}
```
![screenshot](https://github.com/mia-platform/community/assets/10067183/4622e38c-e222-4f6c-aec3-a0bd3fe4bbd2)

```

{
  "level": 50,
  "time": 1686581316853,
  "pid": 6,
  "hostname": "backend-84c6cd65fd-jpvbr",
  "reqId": "b60f1ca4-a0c3-47de-953b-621187700dc0",
  "req": {
    "method": "GET",
    "url": "/api/backend/projects/draft?projectId=newprojectnametestttt&tenantId=tenantid&providerId=github&projectName=newprojectname",
    "hostname": "...",
    "remoteAddress": "127.0.0.1",
    "remotePort": 47922
  },
  "res": { "statusCode": 500 },
  "err": {
    "type": "Error",
    "message": "\"type\" is required!",
    "stack": "Error: \"type\" is required!\n    at $mainenvironmentsVariables (eval at build (/home/node/app/node_modules/fast-json-stringify/index.js:161:20), <anonymous>:1190:15)\n    at $main (eval at build (/home/node/app/node_modules/fast-json-stringify/index.js:161:20), <anonymous>:190:17)\n    at serialize (/home/node/app/node_modules/fastify/lib/reply.js:830:12)\n    at preserializeHookEnd (/home/node/app/node_modules/fastify/lib/reply.js:419:17)\n    at preserializeHook (/home/node/app/node_modules/fastify/lib/reply.js:403:5)\n    at Reply.send (/home/node/app/node_modules/fastify/lib/reply.js:194:7)\n    at /home/node/app/node_modules/fastify/lib/wrapThenable.js:23:15\n    at dynatraceOnServiceExecutionIndicator (/opt/dynatrace/oneagent/agent/bin/1.261.201.20230315-145944/any/nodejs/nodejsagent.js:3111:20)\n    at a.safeInvoke (/opt/dynatrace/oneagent/agent/bin/1.261.201.20230315-145944/any/nodejs/nodejsagent.js:3168:41)\n    at /opt/dynatrace/oneagent/agent/bin/1.261.201.20230315-145944/any/nodejs/nodejsagent.js:8415:25",
    "serialization": { "url": "/api/backend/projects/draft", "method": "GET" }
  },
  "msg": "\"type\" is required!"
}


```

Thanks.

### Actual Outcome

Error "'Type' is required!" is reported.

### Expected Outcome

Continue with the project creation process.

**Comment by fredmaggiowski** [comment]
Hi, there may be an issue with your company setup, is there an empty object configured for `pipelines`, `environmentsVariables` or `staticSecrets`?

**Reply by andredev93** [reply]
Here are the configurations extracted from CMS. There is no `staticSecrets` property, even in the db. Is it a required property? Or where can I find it?

```
pipelines: { type: 'jenkins' }
environmentsVariables: {}
```

**Reply by fredmaggiowski** [reply]
It's fine not to have it, however the environmentVariables empty object seems to be the problem, please open a service request providing the name of the company so that we can assist you

**Reply by andredev93** [reply]
Thanks, opened SR AS-1758.

**Reply by andredev93** [reply]
Looks like by adding the following in the Environment variables field the error does not occur anymore.
```
{
  "type": ""
}
```

I'm not sure this is expected or a bug.

**Reply by fredmaggiowski** [reply, marked as answer]
this is a known issue, not an expected behaviour but still `""` is a worse configuration than `{}`; the proper fix is to unset the property all together but is something that cannot be done by the CMS
